src/dtls.c

```diff
@@ -183,6 +221,7 @@ static int ParseClientHello(const byte* input, word32 helloSz, WolfSSL_CH* ch)
     if (OPAQUE16_LEN + RAN_LEN + OPAQUE8_LEN > helloSz)
         return BUFFER_ERROR;
 
+    ch->msg = input - DTLS_HANDSHAKE_HEADER_SZ;
```

Comment:
I don't think so because this is called inside `DoClientHello` and `DoTls13ClientHello` which receive the buffer already advanced past the header. Do you have any other ideas?

Reply:
You can consider removing the `msg` field from the `WolfSSL_CH` struct. The struct represents the parsed client hello (CH) message, and adding a pointer to the original handshake header may not align with the purpose of the struct.
You may also consider changing its name to `raw` or `handshakeHeader`.
If you remove it you can use an additional parameter to pass the handshake information to `SendStatelessReply`. In both case, I think it's better to set/compute the handshake header address in `DoClientHelloStateless` as you can use `inOutIdx` to be sure to not underflow the input buffer.

Reply:
Changed to use a `raw` pointer and now `Dtls13HashClientHello` recreates the header.